pkg/chartutil/expand.go

```diff
@@ -81,3 +81,67 @@ func ExpandFile(dest, src string) error {
 	defer h.Close()
 	return Expand(dest, h)
 }
+
+// Expand uncompresses and extracts a chart into the specified directory.
+// Return the directory when extracting successfully.
+func ExpandWithPath(dir string, r io.Reader) (string, error) {
+	gr, err := gzip.NewReader(r)
```

Comment:
@FengyunPan I agree about the duplication. What do you think about @bacongobbler's suggestion?